backend/windmill-worker/src/worker.rs

```diff
@@ -1086,163 +1088,96 @@ pub async fn run_worker<R: rsmq_async::RsmqConnection + Send + Sync + Clone + 's
                         } else {
-                            tracing::error!("Failed to fetch script for dedicated worker");
-                            killpill_tx.send(()).expect("send");
-                            return;
+                            tracing::error!(
+                                "flow present but value not found for {}, waiting 10s",
+                                flow_path
+                            );
                         }
+                    } else {
+                        tracing::error!("flow not found for {}, waiting 10s,", flow_path);
                     }
-                    r
-                };
-
-                let worker_envs = build_envs(envs).expect("failed to build envs");
-
-                if let Err(e) = match language {
-                    Some(ScriptLang::Python3) => {
-                        crate::python_executor::start_worker(
-                            lock,
-                            &db,
-                            &content,
-                            &base_internal_url,
-                            &job_dir,
-                            &worker_name,
-                            worker_envs,
-                            &_wp.workspace_id,
-                            &_wp.path,
-                            &token,
-                            job_completed_tx,
-                            dedicated_worker_rx,
-                            killpill_rx,
-                        )
-                        .await
-                    }
-                    Some(ScriptLang::Bun) => {
-                        crate::bun_executor::start_worker(
-                            lock,
-                            &db,
-                            &content,
-                            &base_internal_url,
-                            &job_dir,
-                            &worker_name,
-                            worker_envs,
-                            &_wp.workspace_id,
-                            &_wp.path,
-                            &token,
-                            job_completed_tx,
-                            dedicated_worker_rx,
-                            killpill_rx,
-                        )
-                        .await
-                    }
-                    Some(ScriptLang::Deno) => {
-                        crate::deno_executor::start_worker(
-                            &content,
-                            &base_internal_url,
-                            &job_dir,
-                            &worker_name,
-                            worker_envs,
-                            &_wp.workspace_id,
-                            &_wp.path,
-                            &token,
-                            job_completed_tx,
-                            dedicated_worker_rx,
-                            killpill_rx,
-                        )
-                        .await
+                    tokio::time::sleep(Duration::from_millis(10000)).await;
```

Comment:
Not sure I fully understand why we're spleeping after starting the dedicated worker here 🤔

Reply:
we're sleeping only if we cannot fetch the flow, and looping thereafter